src/common/com/intellij/plugins/haxe/lang/psi/HaxeResolver.java

```diff
@@ -46,8 +46,12 @@
   public static final HaxeResolver INSTANCE = new HaxeResolver();
   public static final String IMPORT_EXTENSION = ".hx";
 
+  public static boolean isExtension = false;
```

Comment:
This really isn't going to work with multi-threaded code.  The annotators run on background threads while the users changes occur on the UI thread.

Perhaps, we could try to add the data to the candidateInfoArray and/or change the signature of the resolve function to add an object that the code would set.  The trouble there is that the resolve caches already expect the current structure and they are part of the IDEA code.

I think that the best way is to duplicate the test that determines whether the isExtension flag should be set, and move it into the models (HaxeType or HaxeTypeResolver, whichever is more appropriate).